File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int sum = 0;
        Map<Integer,Integer> mpp = new HashMap<>();
        mpp.put(0,1);
        int cnt = 0;
        for(int num: nums){
            sum += num;
            if(mpp.containsKey(sum-k)) cnt += mpp.get(sum-k);

            mpp.put(sum, mpp.getOrDefault(sum,0)+1);
        }
        return cnt;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
from collections import defaultdict

class Solution:
    def subarraySum(self, nums, k):
        sum_ = 0
        mpp = defaultdict(int)
        mpp[0] = 1
        cnt = 0

        for num in nums:
            sum_ += num
            if (sum_ - k) in mpp:
                cnt += mpp[sum_ - k]
            mpp[sum_] += 1

        return cnt
